### DISCUSSION TITLE
GraphQL/REST/SDK: is there any principal difference?

### DISCUSSION THREAD

**Discussion by zhibirc** [directus/directus#16889] [Archive]
Hello Directus,

my question in probably too simple, but it's important for me as for novice in Directus land. :)

As far as I understand, for the purpose of communication with Directus we can use one of the following three ways -- GraphQL/REST/SDK, and all of these are "aligned" and equal in their functionality to retrieve data.

I would like to ask what kind of communicational protocols/way is preferred (if any) and has SDK some functionality which is absent or non-trivial using GraphQL/REST (if any)?

Thanks is advance.

**Comment by rijkvanzanten** [comment]
> I would like to ask what kind of communicational protocols/way is preferred (if any) and has SDK some functionality which is absent or non-trivial using GraphQL/REST (if any)?

Up to you! They all support the same functionality set. The SDK is a thin wrapper around axios which calls the regular REST API endpoints, so there's no functional difference between the two.

In general, Directus converts requests coming in (through rest / graphql or soon websockets) into a generic AST which is then parsed through the data pipeline to be queried / modified. That allows all of these interfaces to support the same feature sets and generally have the same performance 🙂
